src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2Solubility.cpp

```diff
@@ -30,337 +28,183 @@ namespace geos
-  // return the reduced volume
-  return CO2EOSSolver::solve( name,
-                              maxNumNewtonIter,
-                              maxNumBacktrackIter,
-                              tolerance,
-                              minAbsDeriv,
-                              maxAbsUpdate,
-                              allowedMinValue,
-                              initialReducedVolume,
-                              T,
-                              P,
-                              presMultiplierForReporting,
-                              f );
+  TableFunction * tableFunction = nullptr;
+  if( functionManager.hasGroup< TableFunction >( tableName ) )
+  {
+    tableFunction = functionManager.getGroupPointer< TableFunction >( tableName );
+  }
+  else
+  {
+    tableFunction = dynamicCast< TableFunction * >( functionManager.createChild( "TableFunction", tableName ) );
+    tableFunction->setTableCoordinates( tableCoords.getCoords(), { units::Pressure, units::TemperatureInC } );
+    tableFunction->setTableValues( values, units::Solubility );
+    tableFunction->setInterpolationMethod( TableFunction::InterpolationType::Linear );
+  }
+  tableFunction->initializeFunction();
+  return tableFunction;
 }
 
-void calculateCO2Solubility( string const & functionName,
-                             real64 const & tolerance,
-                             PTTableCoordinates const & tableCoords,
-                             real64 const & salinity,
-                             array1d< real64 > const & values )
+std::pair< TableFunction const *, TableFunction const * >
+makeSolubilityTables( string const & functionName,
+                      string_array const & inputParams,
+                      constitutive::PVTProps::CO2Solubility::SolubilityModel const & solubilityModel )
 {
-  // Interaction parameters, see Table 2 of Duan and Sun (2003)
-  constexpr real64 mu[] =
-  { 28.9447706, -0.0354581768, -4770.67077, 1.02782768e-5, 33.8126098, 9.04037140e-3,
-    -1.14934031e-3, -0.307405726, -0.0907301486, 9.32713393e-4, 0 };
-  constexpr real64 lambda[] = { -0.411370585, 6.07632013e-4, 97.5347708, 0, 0, 0, 0, -0.0237622469, 0.0170656236, 0, 1.41335834e-5 };
-  constexpr real64 zeta[] = { 3.36389723e-4, -1.98298980e-5, 0, 0, 0, 0, 0, 2.12220830e-3, -5.24873303e-3, 0, 0 };
-
-  localIndex const nPressures = tableCoords.nPressures();
-  localIndex const nTemperatures = tableCoords.nTemperatures();
+  FunctionManager & functionManager = FunctionManager::getInstance();
+  constitutive::PVTProps::PTTableCoordinates tableCoords;
 
-  for( localIndex i = 0; i < nPressures; ++i )
+  // Check solubility model for explicit table input
+  if( solubilityModel == constitutive::PVTProps::CO2Solubility::SolubilityModel::Tables )
   {
-    real64 const P = tableCoords.getPressure( i ) / P_Pa_f;
+    // The default table is a table with all zeros unless the name is explicitly provided
+    tableCoords.appendPressure( 1.0e5 ).appendPressure( 1.0e8 )
```

Comment:
Just out of curiousity/documentation, why these hard-coded values ? Is this all model range ?

Reply:
These are used to create a zero value table. So they need only have a valid range. I have added a comment to that effect.